helix-core/src/main/java/org/apache/helix/model/LiveInstance.java

```diff
@@ -190,6 +192,24 @@ public void setWebserviceUrl(String url) {
     _record.setSimpleField(LiveInstanceProperty.ZKPROPERTYTRANSFERURL.toString(), url);
   }
 
+  /**
+   * Get the current task thread pool size of the instance
+   * @return the current task thread pool size
+   */
+  public int getCurrentTaskThreadPoolSize() {
+    return _record.getIntField(LiveInstanceProperty.CURRENT_TASK_THREAD_POOL_SIZE.name(),
+        CURRENT_TASK_THREAD_POOL_SIZE_NOT_SET);
```

Comment:
Please try to understand the proposed design well. This suggests to me that you should perhaps spend more time trying to understand what this field represents.

For example, what does it mean for an instance to have a task thread pool of size -1? (it is meaningless, we shouldn't create a task pool with size -1).

Reply:
Discussed offline: this is meant to be a bit signaling an error, because field should always be set. But it's true that old participants don't have this field. I'm using the default 40 instead as there's no better options in that case. Thank you.